core/client/src/main/java/alluxio/client/block/stream/BlockInStream.java

```diff
@@ -77,11 +75,9 @@ public static BlockInStream createLocalBlockInStream(long blockId, long blockSiz
     try {
       BlockWorkerClient blockWorkerClient =
           closer.register(context.createBlockWorkerClient(workerNetAddress));
-      LockBlockResource lockBlockResource =
-          closer.register(blockWorkerClient.lockBlock(blockId, LockBlockOptions.defaults()));
       PacketInStream inStream = closer.register(PacketInStream
-          .createLocalPacketInStream(lockBlockResource.getResult().getBlockPath(), blockId,
-              blockSize));
+          .createLocalPacketInStream(context, blockWorkerClient.getDataServerAddress(), blockId,
+              blockWorkerClient.getSessionId(), blockSize));
```

Comment:
I would prefer to get rid of the block worker client interaction all together on the short circuit path in this PR, will it be a significant change?